kitodo-mediaserver-core/src/test/java/org/kitodo/mediaserver/core/services/ActionServiceTest.java

```diff
@@ -0,0 +1,284 @@
+ * LICENSE file that was distributed with this source code.
+ */
+
+package org.kitodo.mediaserver.core.services;
+
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import static org.assertj.core.api.Assertions.assertThat;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.kitodo.mediaserver.core.db.entities.ActionData;
+import org.kitodo.mediaserver.core.db.entities.Work;
+import org.kitodo.mediaserver.core.exceptions.ActionServiceException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.domain.EntityScan;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
+import org.springframework.test.context.junit4.SpringRunner;
+
+/**
+ * Test for the User repository.
+ */
+@SpringBootTest(classes = ActionService.class)
+@EnableJpaRepositories("org.kitodo.mediaserver.core.db.repositories")
+@EntityScan("org.kitodo.mediaserver.core.db.entities")
+@RunWith(SpringRunner.class)
+@ComponentScan(value = "org.kitodo.mediaserver.core.services")
+@DataJpaTest
+public class ActionServiceTest {
+
+    @Autowired
+    private TestEntityManager entityManager;
+
+    @Autowired
+    private ActionService actionService;
+
+    Work work1;
+    Work work2;
+    Map<String, String> parameter1;
+    Map<String, String> parameter2;
+    Map<String, String> parameter3;
+    ActionData actionData1;
+    ActionData actionData2;
+    ActionData actionData3;
+
+    public void init() {
+        work1 = new Work("xyz", "title1");
+        work2 = new Work("abc", "title2");
+
+        parameter1 = new HashMap<>();
+        parameter1.put("key1", "value1");
+
+        parameter2 = new HashMap<>();
+        parameter2.put("key2", "value2");
+
+        parameter3 = new HashMap<>();
+        parameter3.put("key3", "value3");
+
+        actionData1 = new ActionData(work1, "action1", parameter1);
+        actionData2 = new ActionData(work2, "action2", parameter2);
+        actionData3 = new ActionData(work2, "action2", parameter3);
+    }
+
+    public void persist() {
+        entityManager.persist(work1);
+        entityManager.persist(work2);
+        entityManager.persist(actionData1);
+        entityManager.persist(actionData2);
+        entityManager.persist(actionData3);
+    }
+
+    @After
+    public void cleanUp() {
+        for (Object entity : Arrays.asList(actionData1, actionData2, actionData3, work1, work2)) {
+            try {
+                entityManager.remove(entity);
+            } catch (Exception ex) {}
+        }
+    }
+
+    @Test
+    public void requestAction() throws ActionServiceException {
```

Comment:
It is not very clear what is actually being tested. I think it generally would be better, if the test method was named after what is being tested / what is expected rather than after which method is called or what the input is. However, we never really discussed this, so it would probably be good to confront the mattern in a meeting.